### PR TITLE
[TESTS][SQL] Setup testdata at the beginning for tests to run independently

### PR DESCRIPTION
**Pull request by dilipbiswal**
## What changes were proposed in this pull request?

In CachedTableSuite, we are not setting up the test data at the beginning. Some tests fail while trying to run individually. When running the entire suite they run fine.

Here are some of the tests that fail -

- test("SELECT star from cached table")
- test("Self-join cached") 

As part of this simplified a couple of tests by calling a support method to count the number of
InMemoryRelations.

## How was this patch tested?

Ran the failing tests individually.

### CODE REVIEW

**Comment by dilipbiswal** [comment]
retest this please

**Comment by dilipbiswal** [comment]
cc @cloud-fan @gatorsmile Could you please trigger a test for this. Not sure why the last run didn't succeed. Thanks a lot.

**Comment by dilipbiswal** [comment]
ok to test

**Comment by gatorsmile** [comment]
ok to test

**Comment by viirya** [comment]
LGTM

File: sql/core/src/test/scala/org/apache/spark/sql/CachedTableSuite.scala [hunk 1]
```diff
@@ -39,6 +39,8 @@ private case class BigData(s: String)
 class CachedTableSuite extends QueryTest with SQLTestUtils with SharedSQLContext {
   import testImplicits._
 
+  setupTestData
```
Comment (srowen) [inline sql/core/src/test/scala/org/apache/spark/sql/CachedTableSuite.scala]: If it's a method call add parens

Comment (gatorsmile) [inline sql/core/src/test/scala/org/apache/spark/sql/CachedTableSuite.scala]: @dilipbiswal This method changes the value, and thus, we should add parens.

**Comment by gatorsmile** [comment]
@dilipbiswal Since this PR is to fix the minor test issues, could you do me a favor to fix three comments?

Replace `_global_temp` by `global_temp`

- https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/catalog/Catalog.scala#L383-L384
- https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala#L2591-L2592
-
https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/execution/command/views.scala#L50-L51

Thanks

File: sql/core/src/test/scala/org/apache/spark/sql/CachedTableSuite.scala (hunk 1 above)
Comment (dilipbiswal) [inline sql/core/src/test/scala/org/apache/spark/sql/CachedTableSuite.scala]: OK.. will do.

**Comment by dilipbiswal** [comment]
@gatorsmile Sure Sean. Will do

**Comment by gatorsmile** [comment]
LGTM

**Comment by gatorsmile** [comment]
Thanks! Merging to master.

**Comment by dilipbiswal** [comment]
@gatorsmile @srowen @viirya Thank you very much !!

**Pull request closed without merging**